src/client/line.cpp

```diff
@@ -0,0 +1,547 @@
+			{
+				if (params->last_changed_props[prop])
+					switch ((LineParams::LineProperty)prop)
+					{
+						case LineParams::LP_TYPE:
+							line_params.type = params->type;
+							break;
+						case LineParams::LP_START_POS:
+							line_params.pos.first = params->pos.first;
+							break;
+						case LineParams::LP_END_POS:
+							line_params.pos.second = params->pos.second;
+							break;
+						case LineParams::LP_START_COLOR:
+							line_params.color.first = params->color.first;
+							break;
+						case LineParams::LP_END_COLOR:
+							line_params.color.second = params->color.second;
+							break;
+						case LineParams::LP_START_ATTACH_TO:
+							line_params.attached_ids.first = params->attached_ids.first;
+							break;
+						case LineParams::LP_END_ATTACH_TO:
+							line_params.attached_ids.second = params->attached_ids.second;
+							break;
+						case LineParams::LP_WIDTH:
+							line_params.width = params->width;
+							break;
+						case LineParams::LP_TEXTURE:
+							line_params.texture = params->texture;
+							m_line_scenenode_list[id]->updateMaterial();
+							break;
+						case LineParams::LP_CLAMP_MODE:
+							line_params.clamp_mode = params->clamp_mode;
+							m_line_scenenode_list[id]->updateMaterial();
+							break;
+						case LineParams::LP_PLAYERNAME:
+							line_params.playername = params->playername;
+							break;
+						case LineParams::LP_ANIMATION:
+							line_params.animation = params->animation;
+							break;
+						case LineParams::LP_BACKFACE_CULLING:
+							line_params.backface_culling = params->backface_culling;
+							m_line_scenenode_list[id]->updateMaterial();
+							break;
+						case LineParams::LP_LIGHT_LEVEL:
+							line_params.light_level = params->light_level;
+							break;
+						case LineParams::LP_AXIS_ANGLE:
+							line_params.axis_angle = params->axis_angle;
+							break;
+					}
+			}
+			break;
+		}
+		case CE_REMOVE_3DLINE:
+		{
+			remove3DLineFromList(id);
+			break;
+		}
+	}
+
+	delete params;
+}
+
+void LineSceneNodeManager::step(f32 dtime)
+{
+	static thread_local const float radius =
+			g_settings->getS16("max_block_send_distance") * MAP_BLOCKSIZE * BS;
+
+	v3f ppos = m_env->getLocalPlayer()->getPosition();
+
+	for (auto line : m_line_scenenode_list)
+	{
+		auto& pos = line.second->m_pos;
+
+		v3f cam_offset = intToFloat(m_env->getCameraOffset(), BS);
+		bool is_far_from_start_pos = ppos.getDistanceFromSQ(pos.first+cam_offset) > radius * radius;
+		bool is_far_from_end_pos = ppos.getDistanceFromSQ(pos.second+cam_offset) > radius * radius;
+
+		bool is_attaches_invalid = false;
+
+		if (line.second->m_params.attached_ids.first != 0)
+		{
+			if (not m_env->getGenericCAO(line.second->m_params.attached_ids.first))
+				is_attaches_invalid = true;
+		}
+
+		if (line.second->m_params.attached_ids.second != 0)
+		{
+			if (not m_env->getGenericCAO(line.second->m_params.attached_ids.second))
+				is_attaches_invalid = is_attaches_invalid || true;
+		}
```

Comment:
Please do not use alternative tokens. See [Code style guidelines#Miscellaneous](https://dev.minetest.net/Code_style_guidelines#Miscellaneous) for reference.

Reply:
While we're at it:
1. `if (var != 0)` could be shortened to `if (var)` (at at least OK for pointers and integers)
2. `is_attaches_invalid = is_attaches_invalid || true;` can be simplified to `= true`
3. rename suggestion: `is_attaches_invalid` to `is_attachment_invalid`

Reply:
Fixed.